### DISCUSSION TITLE
LAB-D Prefetch 流程詢問

### DISCUSSION THREAD

**Discussion by Yanlin900414** [bol-edu/HLS-SOC-Discussions#174] [Q&A] [answered]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145038188/47bd89af-7211-4a83-8b74-775fa45a91ec)
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145038188/cb555f90-bc54-4bc3-9e1b-9901083add63)

不好意思，想詢問一下各位，目前正在完成Prefetch的部分，若如上兩張圖的流程來看，每次READ都會有個3個T的等待時間，READ_RES也會有3個T的等待時間，若是要製作Prefetch的話讀取兩筆資料到Buffer中，就要進行IDLE(6)->READ(9)->READ_RES(10)，這樣與原先一筆一筆資料讀取出來的時間差不多(下方圖)，這樣進行Prefetch的意義不大，因此這部分感到非常疑惑，是否有其他人有不一樣的想法呢，謝謝 !
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145038188/b7ce1b38-f21d-4d60-a2d0-45221f95304f)

**Comment by alankuo04** [comment, marked as answer]
同學你好

第一次進行READ的時候會需要先進行ACTIVATE，這個部分需要有3個T的等待時間，而在第二次如果讀取的是同一個BANK時，則只需要3個T的READ來讀取資料就可以了，READ_RES所花的時間都只有1個T

Prefetch的部分可以看到在controller中有一個busy的訊號，在接收到第一筆READ command後很快就從1變為0，這時候可以再對controller傳送下一個位置的READ command，在有進行prefetch時 IDLE(6) 這個state應該只會持續1個T而已

**Reply by Yanlin900414** [reply]
謝謝您! 我發現我打錯了 READ_RES確實只花一個T

**Comment by boledulab** [comment]
Further, next host read does not come immediately. DRAM can prefetch its data during the host idle time, so that when host request comes, it can immediately return the data. Do make sure the next host request is to the data prefetched.
